cmd/local-dev/setup.go

```diff
@@ -17,21 +17,46 @@ package main
 
 import (
 	"bytes"
+	"context"
 	"fmt"
 	"io/fs"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
+const (
+	localRegistryName = "registry.local"
+	localRegistryPort = 5001
+)
+
+var kindClusterConfig = `
+apiVersion: kind.x-k8s.io/v1alpha4
+kind: Cluster
+name: "%s"
+nodes:
+- role: control-plane
+  image: "%s"
+# Configure registry for KinD.
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."%s:%d"]
+    endpoint = ["http://%s:%d"]
+`
+
 func addSetupFlags(cmd *cobra.Command) {
 	cmd.Flags().String("cluster-name", "policy-controller-demo", "name of the dev policy controller cluster")
-	cmd.Flags().String("ko-docker-repo", "", "name of the Ko Docker repository to use")
-	cmd.MarkFlagRequired("ko-docker-repo") //nolint:errcheck
+	cmd.Flags().String("registry-url", "registry.local", "URL of the Ko Docker registry to use. If no registry is provided, the local Kind registry will be used")
+	cmd.Flags().String("k8s-version", "v1.26.x", "name of the Ko Docker repository to use")
```

Comment:
If we use `v1.x.x` format, we should explain the other valid values. It isn't clear to me which is a valid k8s version. Perhaps we should use regexp to valid the chosen version.